source/common/network/address_impl.cc

```diff
@@ -160,5 +160,49 @@ InstanceConstSharedPtr parseInternetAddress(const std::string& ip_addr) {
   return nullptr;
 }
 
+InstanceConstSharedPtr parseInternetAddressAndPort(const std::string& addr) {
```

Comment:
I assume you already did this, but please check full coverage of all the error cases.